kore/src/Kore/Step/Simplification/InjSimplifier.hs

```diff
@@ -165,7 +191,20 @@ mkInjSimplifier sortGraph =
                 child2' = injChild inj2
             pure (Pair child1' child2' <$ inj2)
 
-      | otherwise = Nothing
+      -- If the child patterns are simplifiable, then they could eventually be
+      -- simplified to produce matching sort injections, but if they are
+      -- non-simplifiable, then they will never match.
+      | hasConstructorLikeTop (injChild inj1) = Left Distinct
+      | hasConstructorLikeTop (injChild inj2) = Left Distinct
+
+      -- Even if the child patterns are simplifiable, if they do not have any
+      -- common subsorts, then they will never simplify to produce matching sort
+      -- injections.
```

Comment:
We do assume both (1) and (2). The frontend does give us the complete sort graph, but we technically lack the no-confusion axioms required to determine distinctness of sorts. Regarding (2) specifically: because Kore is many-sorted and K is order-sorted, the frontend must emit a distinct (Kore) sort for every (K) sort intersection.